org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/core/ReplModule.java

```diff
@@ -25,9 +26,11 @@
 import com.google.inject.multibindings.MapBinder;
 
 /**
- * Client library bindings.
+ * This class binds the core classes. It is intended to be subclassed by client implementations.
+ * These subclasses should bind their implementations of {@link IRepl}, {@link IMessageHook} and
+ * {@link IResultHook}.
  */
-public class ReplModule extends SpoofaxModule {
+public abstract class ReplModule extends SpoofaxModule {
```

Comment:
Instead of an abstract class we can also provide dummies for the interfaces. I think this is a better solution.